gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/ManagedCleanableDatasetFinder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention.profile;
+
+import gobblin.config.client.ConfigClient;
+import gobblin.config.client.api.ConfigStoreFactoryDoesNotExistsException;
+import gobblin.config.client.api.VersionStabilityPolicy;
+import gobblin.config.store.api.ConfigStoreCreationException;
+import gobblin.config.store.api.VersionDoesNotExistException;
+import gobblin.data.management.retention.ManagedDatasetCleaner;
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.retention.version.DatasetVersion;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.typesafe.config.Config;
+
+
+/**
+ * A {@link ConfigurableGlobDatasetFinder} backed by gobblin-config-management. It uses {@link ConfigClient} to get dataset configs
+ */
+public class ManagedCleanableDatasetFinder extends
+    ConfigurableGlobDatasetFinder<ConfigurableCleanableDataset<DatasetVersion>> {
+
+  private final ConfigClient client;
+
+  public ManagedCleanableDatasetFinder(FileSystem fs, Properties jobProps, Config config) throws IOException {
+    super(fs, jobProps, config);
+    this.client = ConfigClient.createConfigClient(VersionStabilityPolicy.STRONG_LOCAL_STABILITY);
```

Comment:
The configClient should be only create once. Otherwise, we may hit the race condition of new config store version . Also the cache is not shared across the ConfigClient